### PR TITLE
chore: code cleanup

### PR DESCRIPTION
**Pull request by feedm3**
This PR restructures our code:
- all gatsby related SSG code is now in `/gatsby`
- `create-pages` functions all have their own file
- deleted `client-pages` because it wasn't used and is also outdated
- removed `team.json` because it was outdated and unused

It also contains a bug fix:
- adapt redirect urls

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -1,30 +1,16 @@
...
- *
- * Only lower case letters and numbers are used, everything else is filtered out.
- *
- */
-const extractGatsbyCloudPreviewUrl = () => {
-  const PRODUCTION_BRANCH_NAME = 'main';
-  const DOMAIN_NAME = 'satellytescommain';
+const {
+  detectGatsbyCloudPreviewUrl,
+} = require('./gatsby/util/detect-gatsby-cloud-preview-url');
```
Comment (georgiee) [inline gatsby-config.js]: nice, that's how I imagined a refactoring for the gatsby specific files 👍

File: gatsby-node.js [hunk 2]
```diff
@@ -1,351 +1,35 @@
...
+exports.onCreateNode = (gatsbyCreateNodeArgs) => {
+  createPreviewCard(gatsbyCreateNodeArgs);
 };
 
 exports.createPages = async (createPagesArgs) => {
-  await createCareerPages(createPagesArgs);
-  await createBlogPages(createPagesArgs);
-  // await createClientPages(createPagesArgs);
-
+  await createCareerDetails(createPagesArgs);
```
Comment (georgiee) [inline gatsby-node.js:28]: better name 👍

File: gatsby/create-pages/create-blog-posts.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const { appendTrailingSlash } = require('../util/append-trailing-slash');
+const path = require('path');
+
+const BLOG_POST_TEMPLATE_PATH = path.resolve('src/templates/blog-post.tsx');
+
+const createBlogPosts = async ({ actions, reporter, graphql }) => {
```
Comment (georgiee) [inline gatsby/create-pages/create-blog-posts.js:6]: much easier to focus and to read now 🌟

File: gatsby/create-pages/create-carrer-details.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+
+/**
+ * Generate all career pages by querying the personio API.
+ * Currently this extracts a single (default) language even though we can
+ * author multiple languages.
+ *
+ * The query will also create a matching social card in `public/`
+ * and pass it to the context of the created page.
+ */
+const createCareerDetails = async ({ actions }) => {
```
Comment (georgiee) [inline gatsby/create-pages/create-carrer-details.js:33]: similar to the blog post function, much easier to focus on the content now.

File: gatsby/create-pages/create-carrer-details.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+const { appendTrailingSlash } = require('../util/append-trailing-slash');
+
+const CAREER_TEMPLATE_PATH = path.resolve('src/templates/career.tsx');
+const CAREER_DETAILS_TEMPLATE_PATH = path.resolve(
+  'src/templates/career-details.tsx',
+);
+
+const PERSONIO_JOBS_URL = 'https://satellytes.jobs.personio.de/xml';
+const PERSONIO_SHORT_DESCRIPTION_NAME = 'Kurzbeschreibung';
+const LANGUAGES = ['en', 'de'];
```
Comment (georgiee) [inline gatsby/create-pages/create-carrer-details.js]: I expected this part to be relevant for other content too, how does it come that we only need that list here?

File: gatsby/create-pages/create-redirects.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/**
+ * Sometimes we mess up URLs. Fixing them is necessary, but if Google or other
+ * Link on external pages still link to the old URL they will only get the 404
+ * page. That is why we need to redirect the old links to the new ones.
+ */
+const createRedirects = ({ actions }) => {
+  const { createRedirect } = actions;
+
+  const redirects = [
```
Comment (georgiee) [inline gatsby/create-pages/create-redirects.js]: the previous version was this one. Don't we need the `/career/325433-senior-backend-engineer-(mwx)/` variant to be redirected? Or wasn't it indexed before? 
```
const redirects = [
    {
      fromPath: '/career/325433-senior-backend-engineer-m-w-x/',
      toPath: '/career/325433-senior-backend-engineer-(mwx)/',
    },
    {
      fromPath: '/career/325420-senior-frontend-engineer-m-w-x/',
      toPath: '/career/325420-senior-frontend-engineer-(mwx)/',
    },
    {
      fromPath: '/career/338504-frontend-engineer-w-m-x/',
      toPath: '/career/338504-frontend-engineer-(wmx)/',
    },
    {
      fromPath: '/career/325493-senior-uxui-designer-m-w-x/',
      toPath: '/career/325493-senior-uxui-designer-(mwx)/',
    },
    {
      fromPath: '/page/imprint/',
      toPath: '/imprint/',
    },
  ];
```

File: gatsby/util/append-trailing-slash.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+const appendTrailingSlash = (path) => {
```
Comment (georgiee) [inline gatsby/util/append-trailing-slash.js:1]: having all those isolated files brings testing back to the table and we can add that either with this refactoring PR or in a later PR. I think testing the gatsby code is currently more important than testing the components on our website.

**Review by georgiee — APPROVED** [review]
I think our files grow very well within the single files of gatsby. The new file structure to hold our gatsby configuration scripts makes absolutely sense. Throw some tests in there and we have a solid environment to extend in the future 🙌

I have never seen anyone talking about splitting the content of gatsby node into separate files. Do people just append any change without even considering a split ?


```
gatsby
├── create-node
│   └── create-preview-card.js
├── create-pages
│   ├── create-blog-posts.js
│   ├── create-carrer-details.js
│   └── create-redirects.js
└── util
    ├── append-trailing-slash.js
    ├── detect-gatsby-cloud-preview-url.js
    ├── extract-personio-slug.js
    ├── extract-personio-slug.test.js
    └── preview-card-generator
```
I like the structure 👍 

Do we want some tests for crucial utils or methods? Do you have anything fragile in mind? If not let's merge otherwise let's add a few tests. Thank you for the refactoring.

FYI: I noticed you use `chore`, but `refactor` is common commit type so you could use that instead too.

File: gatsby/create-pages/create-carrer-details.js (hunk 5 above)
Comment (feedm3) [reply gatsby/create-pages/create-carrer-details.js]: Yep, we also use the languages in the gatsby-config. I will move it there to only have one supported languages field

File: gatsby/create-pages/create-redirects.js (hunk 6 above)
Comment (feedm3) [reply gatsby/create-pages/create-redirects.js]: You are right. We also need to redirect to "original" path now. Will add it. Good catch :)

File: gatsby/util/append-trailing-slash.js (hunk 7 above)
Comment (feedm3) [reply gatsby/util/append-trailing-slash.js:1]: Yes, those isolated files encourage a good architecture and design - which in turn makes testing quite simple. I will add a test for this one.

Comment (feedm3) [reply gatsby/util/append-trailing-slash.js:1]: I also added a test for `build-gatsby-cloud-preview-url.js`

File: gatsby/create-pages/create-redirects.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
+const REDIRECTS = [
```
Comment (georgiee) [inline gatsby/create-pages/create-redirects.js:1]: We can probably remove those redirects in a few weeks once the search indices are updated? We have very low traffic anyway.

File: gatsby/util/append-trailing-slash.test.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+const { appendTrailingSlash } = require('./append-trailing-slash');
+
+describe('appendTrailingSlash', () => {
```
Comment (georgiee) [inline gatsby/util/append-trailing-slash.test.js:3]: 🙌

File: gatsby/util/build-gatsby-cloud-preview-url.test.js [hunk 10]
```diff
@@ -0,0 +1,27 @@
+const {
+  buildGatsbyCloudPreviewUrl,
+} = require('./build-gatsby-cloud-preview-url');
+
+describe('build-gatsby-cloud-preview-url', () => {
```
Comment (georgiee) [inline gatsby/util/build-gatsby-cloud-preview-url.test.js:5]: every test I see here makes the code base feel much less a house of cards 🥳

**Review by georgiee — APPROVED** [review]
thanks for the the last commits with the added tests

File: gatsby/create-pages/create-redirects.js (hunk 8 above)
Comment (feedm3) [reply gatsby/create-pages/create-redirects.js:1]: Yep, let's check the search console from time to time and clean this up.

**Pull request merged**